packages/auth/src/types/Auth.ts

```diff
@@ -91,8 +91,11 @@ export function isFederatedSignInOptionsCustom(
 	return obj && !!keys.find(k => obj.hasOwnProperty(k));
 }
 
-export function hasCustomState(obj: any): obj is FederatedSignInOptions {
-	const keys: (keyof FederatedSignInOptions)[] = ['customState'];
+export function hasCustomState(obj: any): boolean {
+	const keys: (
+		| keyof FederatedSignInOptions
+		| keyof FederatedSignInOptionsCustom
+	)[] = ['customState'];
```

Comment:
Your change works fine, I'd suggest narrowing down the type further like this: (so the resulting type is `const keys: "customState"[]` instead of `const keys: ("provider" | "customState" | "customProvider")[]`)

```suggestion
	const keys: (keyof (
		| FederatedSignInOptions
		| FederatedSignInOptionsCustom
	))[] = ['customState'];
```